app/code/Magento/NewsletterApi/Api/Data/SubscriptionInterface.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NewsletterApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+use Magento\NewsletterApi\Api\Data\SubscriptionExtensionInterface;
+
+/**
+ * Newsletter Subscription Interface
+ *
+ * represents one newsletter subscription identified by an
+ * e-mail address or the internal id
+ *
+ * @api
+ */
+interface SubscriptionInterface extends ExtensibleDataInterface
+{
+    /**#@+
+     * Constants for keys of data array. Identical to the name of the getter in snake case
+     */
+    const SUBSCRIPTION_ID = 'subscription_id';
+    const EMAIL = 'email';
+    const STATUS = 'status';
+    /**#@-*/
+
+    /**
+     * Set ID
+     *
+     * set the internal id of the entity
+     *
+     * @param int $id
+     * @return void
+     */
+    public function setId($id);
+
+    /**
+     * Set E-Mail Address
+     *
+     * set the e-mail address of the subscriber
+     *
+     * @param string $email
+     * @return void
+     */
+    public function setEmail(string $email);
+
+    /**
+     * Set Status
+     *
+     * set the status of the subscription
+     *
+     * @param int $status
```

Comment:
if so, on the level of API we have to provide the ability to extend the status list. 
And provide a meaningful way of moving the state between all of these statuses. Like vending machine. [State](https://sourcemaking.com/design_patterns/state) design pattern could be used here if needed.
But that should be declared explicitly enough on the level of APIs.
Because if we leave that as implementation details. Some could want to extend list of statuses and add custom one. For these purposes he/she will have to extend/customize private implementation, but not the API contract, because such extension point is not provided